packages/sdk-socket-server-next/src/analytics-api.ts

```diff
@@ -314,22 +314,30 @@ app.post('/evt', async (_req, res) => {
       },
     };
 
-    // Replace 'sdk' id wichch translates to '5a374dcd2e5eb762b527af3a5bab6072a4d24493' with a unique random id
-    if (event.userId === SDK_EXTENSION_DEFAULT_ID) {
+    // Always check for userId to avoid hot sharding events
+    if (!event.userId) {
       const newUserId = uuidv4();
-      logger.info(`Replacing 'sdk' id with a random uuid=${newUserId}`);
+      logger.info(
+        `event: ${event.event} - Replacing 'sdk' id with '${newUserId}'`,
```

Comment:
Won't this really spam our logs?
Should we not properly metric those instead, as there is almost no cardinality in this message?
Or have it as debug logging as otherwise we just get flooded with thousands of lines.

Reply:
debug logging won't help and wont be displayed anyway

Reply:
but good point, keeping it just in case if we deploy to dev

Reply:
yes aware, but if we are having trouble, instead of code changes, it would just be changing an env variable in the deployment?

Reply:
yes